lib/perl/Genome/Model/Build.pm

```diff
@@ -1395,6 +1395,11 @@ sub _launch {
     local $ENV{UR_DUMP_STATUS_MESSAGES} = 1;
     local $ENV{UR_COMMAND_DUMP_STATUS_MESSAGES} = 1;
 
+    # we don't use 'local' because we want this to persist through
+    # spawned workflows
+    $ENV{Genome::Config::spec('lsf_project_name')->env} =
+        sprintf("build/%s", $self->id);
```

Comment:
I've tested this out, and because of the scope-guard that `Genome::Config::set_env` the value does not make it into the environment of the workflow steps.  The same is true of the `local` modifier.  As for nested builds (something that we should eliminate if they exist imo) if they are started by calling `Genome::Model::Build->start()` or something that invokes that then they will behave as expected and have the `projectName` associated with their build and not their parent's build.